### ISSUE TITLE
Facepile block conflict with Webmention v5.3.0

### ISSUE THREAD

**Issue by cagrimmett** [janboddez/indieblocks#180]
Hi! After [Webmention updated to v5.3.0](https://github.com/pfefferle/wordpress-webmention/releases/tag/5.3.0) this morning, I got a fatal on my single templates. Tracked it down to the Facepile block. Temporarily removed the facepile block from my single templates. 

Stack trace:

```
PHP Fatal error: Uncaught Error: Attempt to modify property "query_vars" on string in /srv/htdocs/wp-content/plugins/webmention/includes/class-comment-walker.php:307 
Stack trace: 
#0 /wordpress/core/6.5.3/wp-includes/class-wp-hook.php(324): Webmention\Comment_Walker::comment_query('AND') 
#1 /wordpress/core/6.5.3/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array) 
#2 /wordpress/core/6.5.3/wp-includes/plugin.php(565): WP_Hook->do_action(Array) 
#3 /wordpress/core/6.5.3/wp-includes/class-wp-comment-query.php(394): do_action_ref_array('pre_get_comment...', Array) 
#4 /wordpress/core/6.5.3/wp-includes/class-wp-comment-query.php(366): WP_Comment_Query->get_comments() 
#5 /wordpress/core/6.5.3/wp-includes/class-wp-comment-query.php(323): WP_Comment_Query->query(Array) 
#6 /srv/htdocs/wp-content/plugins/indieblocks/includes/functions.php(418): WP_Comment_Query->__construct(Array) 
#7 /srv/htdocs/wp-content/plugins/indieblocks/blocks/facepile-content/class-facepile-content-block.php(36): IndieBlocks\get_facepile_comments(9781) 
#8 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(463): IndieBlocks\Facepile_Content_Block::render_facepile_content_block(Array, '', Object(WP_Block)) 
#9 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(443): WP_Block->render() 
#10 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(443): WP_Block->render() 
#11 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(443): WP_Block->render() 
#12 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(443): WP_Block->render() 
#13 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(443): WP_Block->render() 
#14 /wordpress/core/6.5.3/wp-includes/class-wp-block.php(443): WP_Block->render() 
#15 /wordpress/core/6.5.3/wp-includes/blocks.php(1705): WP_Block->render() 
#16 /wordpress/core/6.5.3/wp-includes/blocks.php(1743): render_block(Array) 
#17 /wordpress/core/6.5.3/wp-includes/block-template.php(260): do_blocks('<!-- wp:group {...') 
#18 /wordpress/core/6.5.3/wp-includes/template-canvas.php(12): get_the_block_template_html() 
#19 /wordpress/core/6.5.3/wp-includes/template-loader.php(106): include('/wordpress/core...') 
#20 /wordpress/core/6.5.3/wp-blog-header.php(19): require_once('/wordpress/core...') 
#21 /wordpress/core/6.5.3/index.php(17): require('/wordpress/core...') 
#22 {main} thrown in /srv/htdocs/wp-content/plugins/webmention/includes/class-comment-walker.php on line 307
```

**Comment by janboddez** [comment]
No clue why `$query` would suddenly be a string (in `Attempt to modify property "query_vars" on string in /srv/htdocs/wp-content/plugins/webmention/includes/class-comment-walker.php:30`) ...

If you deselect Webmention's "separate facepile" option (or whatever it's called), does it work as before? Because the filter seems active only when that option is enabled.

If you use the Facepile block (but also core's Comments block), and have IndieBlocks' facepile option enabled, you shouldn't need Webmention to filter the comments; IndieBlocks should handle that already.

Update: Oh, wait ... looks like `Webmention\Comment_Walker::comment_query('AND')` is passed a string equal to `AND` rather then a WP_Comment_Query object ... That looks a lot like the `AND` right here (which should be superfluous anyway): https://github.com/janboddez/indieblocks/blob/a2d59de358031056a649ee47a1332ce9e39d4ce2/includes/functions.php#L405

**Comment by janboddez** [comment]
Bug must be here:
https://github.com/pfefferle/wordpress-webmention/blob/a9129f46da7c11876e8ff1a18ebfd90b332eaf81/includes/class-comment-walker.php#L295

**Comment by janboddez** [comment]
Created a PR.

**Comment by janboddez** [comment]
Haha, looks like it was already being addressed. Should be fixed in v5.3.1.
